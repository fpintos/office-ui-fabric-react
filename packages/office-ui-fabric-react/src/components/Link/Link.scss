@import '../../common/common';

// Mixins: to be shared with Link.Example.scss
@mixin root {
  color: $ms-color-themePrimary;
  margin: 0;
  overflow: inherit;
  padding: 0;
  text-overflow: inherit;
}

@mixin isEnabled {
  &:active,
  &:hover,
  &:active:hover {
    color: $ms-color-themeDarker;
  }

  &:focus {
    color: $ms-color-themePrimary;
  }
}

@mixin isDisabled {
  color: $ms-color-neutralTertiary;
  pointer-events: none;
  cursor: default;
}

// Element: ms-Link root component
.root {
  @include root;
}

// State: The button is not disabled.
.isEnabled {
  @include isEnabled;
}

// State: The button is disabled.
.isDisabled {
  @include isDisabled;
}

// Link implemented as a button.
button.root {
  @include focus-border;
  @include text-align(left);
  background: none;
  border: none;
  cursor: pointer;
  display: inline;
  font-size: inherit; // Ensure that we inherit font size, rather than the browser's default.

  @include high-contrast {
    // Using Highlight here so we still have a themeable system color that is different from both window text and link text.
    color: Highlight;
  }
}

// Link implemented as an anchor.
a.root {
  @include focus-outline;
  text-decoration: none;
}
