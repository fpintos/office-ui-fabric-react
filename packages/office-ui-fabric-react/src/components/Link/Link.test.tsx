import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { Link } from './Link';

describe('Link', () => {
  it('renders Link correctly', () => {
    const component = renderer.create(<Link href='#'>I'm a link</Link>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders disabled Link correctly', () => {
    const component = renderer.create(<Link href='#' disabled={ true }>I'm a disabled link</Link>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders Link with no href as a button', () => {
    const component = renderer.create(<Link>I'm a link as a button</Link>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders disabled Link with no href as a button correctly', () => {
    const component = renderer.create(<Link disabled={ true }>I'm a link as a button</Link>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});