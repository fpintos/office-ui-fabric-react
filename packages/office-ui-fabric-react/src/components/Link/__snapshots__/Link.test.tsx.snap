// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders Link correctly 1`] = `
<a
  aria-disabled={undefined}
  className="ms-Link"
  href="#"
  onClick={[Function]}
  target={undefined}
>
  I'm a link
</a>
`;

exports[`Link renders Link with no href as a button 1`] = `
<button
  aria-disabled={undefined}
  className="ms-Link"
  onClick={[Function]}
>
  I'm a link as a button
</button>
`;

exports[`Link renders disabled Link correctly 1`] = `
<a
  aria-disabled={true}
  className="ms-Link is-disabled undefined"
  href="#"
  onClick={[Function]}
  target={undefined}
>
  I'm a disabled link
</a>
`;

exports[`Link renders disabled Link with no href as a button correctly 1`] = `
<button
  aria-disabled={true}
  className="ms-Link is-disabled undefined"
  disabled={true}
  onClick={[Function]}
>
  I'm a link as a button
</button>
`;
